Sources/Masse/MP3Duration.swift

```diff
@@ -0,0 +1,349 @@
+/// Read MP3 files and parse the headers in order to calculate the duration of the
+/// MP3 file.
+/// Supports constant bitrate and variable bitrate
+/// Links:
+/// - [Format reference](https://www.codeproject.com/Articles/8295/MPEG-Audio-Frame-Header)
+/// - [Audio test file source](http://freemusicarchive.org/music/Karine_Gilanyan/Beethovens_Sonata_No_15_in_D_Major/Beethoven_-_Piano_Sonata_nr15_in_D_major_op28_Pastoral_-_I_Allegro)
+/// - [Non-audio test file source](https://www.pexels.com/photo/close-up-of-tiled-floor-258805/)
+
+
+/// Errors that can happen during reading the input stream
+enum InputStreamError: Error {
+    case endOfBuffer
+    case streamError(Error?)
+}
+
+extension InputStream {
+    /// Read `length` into a buffer. Throw a `InputStreamError` on failure
+    func readInto(buffer: UnsafeMutablePointer<UInt8>, length: Int) throws {
+        switch self.read(buffer, maxLength: length) {
+        case 0: throw InputStreamError.endOfBuffer
+        case -1: throw InputStreamError.streamError(self.streamError)
+        default: ()
+        }
+    }
+}
+
+/// Various errors that can happen during MP3 decoding
+/// Especially for invalid MP3 files
+enum MP3DurationError: Error {
+    case invalidFile(URL)
+    case forbiddenVersion(UInt32)
+    case forbiddenLayer
+    case forbiddenMode
+    case invalidBitrate(Int)
+    case invalidSamplingRate(Int)
+    case unexpectedFrame(Int)
+    case readError(Error)
+}
+
+/// Lightweight wrapper around the seconds and nanoseconds
+/// that are encoded in an MP3 file
+public struct Duration {
+    private var seconds: Double
+    
+    public var asSeconds: Double {
```

Comment:
What about naming these `seconds` and `minutes` respectively? The private `seconds` could be renamed to `_seconds` to avoid a naming collision.